app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt

```diff
@@ -12,19 +15,41 @@ import javax.inject.Inject
 typealias Ui = EnterOtpScreen
 typealias UiChange = (Ui) -> Unit
 
+private const val OTP_LENGTH = 6
+
 class EnterOtpScreenController @Inject constructor(
-    private val userSession: UserSession
+    private val userSession: UserSession,
+    private val syncScheduler: SyncScheduler
 ) : ObservableTransformer<UiEvent, UiChange> {
 
   override fun apply(events: Observable<UiEvent>): Observable<UiChange> {
     val replayedEvents = events.compose(ReportAnalyticsEvents()).replay().refCount()
 
     return Observable.merge(
         showPhoneNumberOnStart(replayedEvents),
-        handleBackClicks(replayedEvents)
+        handleBackClicks(replayedEvents),
+        showOtpValidationErrors(replayedEvents),
+        handleOtpSubmitted(replayedEvents)
     )
   }
 
+  private fun showOtpValidationErrors(events: Observable<UiEvent>): Observable<UiChange> {
+    val showIncorrectOtpErrors = events.ofType<EnterOtpSubmitted>()
+        .map { it.otp.length }
+        .map {
+          if (it != OTP_LENGTH) {
+            { ui: Ui -> ui.showIncorrectOtpError() }
+          } else {
+            { _ -> }
```

Comment:
If you don't want to do anything, can we just use a filter here? :p

Reply:
Right, sometimes my brain doesn't function.